doc/advanced_configuration.rst

```diff
@@ -51,10 +51,10 @@ you would do:
 .. code-block:: python
 
     sphinx_gallery_conf = {
-        'filename_pattern' : 'plot_awesome_example.py'
+        'filename_pattern' : 'plot_awesome_example\.py'
     }
 
-Similarly, to build only examples in a specific directory, you can do:
+Here, we escape ``'.'`` because it is a special character in `regular expressions`_. Similarly, to build only examples in a specific directory, you can do:
```

Comment:
how about?
Here, one should escape the dot ``'\.'`` otherwise python `regular expressions`_ matches any character. Nevertheless  as one is targeting a specific file, it is most certainly going to match the dot in the filename.